### PR TITLE
Add `@` shorthand for `CoordinateSystem` methods `coords_to_point` (`c2p`) and `point_to_coords` (`p2c`)

### PR DESCRIPTION
**Pull request by JasonGrace2282**
Allows `ax @ (1, 0, 0)` for `coords_to_point` and `(1, 0, 0) @ ax` for `point_to_coords`

Inspired by @abul4fia

### CODE REVIEW

**Review by behackl — APPROVED** [review]
I like this a lot! 👍 

It might make sense modify or include a new example in the example gallery that uses this notation.

**Review by Viicos — CHANGES_REQUESTED** [review]

File: manim/mobject/graphing/coordinate_systems.py [hunk 1]
```diff
@@ -1779,6 +1780,14 @@ def construct(self):
 
         return T_label_group
 
+    def __matmul__(self, coord: Point3D | Mobject):
+        if isinstance(coord, Mobject):
+            coord = coord.get_center()
+        return self.coords_to_point(*coord)
```
[the remark this answers is not in the document]
Comment (JasonGrace2282) [reply manim/mobject/graphing/coordinate_systems.py:1786]: The problem with removing the `*` is that `ax @ (1, 0, 0)` then returns a 3x3 array, which I felt was kinda unintuitive.

**Comment by JasonGrace2282** [comment]
I just added it to `NumberLine` and changed the `CoordinateSystem` example to use `@` notation. In the future, maybe we should transition to that being the "standard" (although `c2p` can always exist for more complex expressions)

**Comment by Viicos** [comment]
Will get back to the unresolved comment soon

File: manim/mobject/graphing/coordinate_systems.py [hunk 2]
```diff
@@ -144,7 +145,7 @@ def __init__(
         self.y_length = y_length
         self.num_sampled_graph_points_per_tick = 10
 
-    def coords_to_point(self, *coords: Sequence[ManimFloat]):
+    def coords_to_point(self, *coords: Point3D):
```
Comment (Viicos) [inline manim/mobject/graphing/coordinate_systems.py:149]:
```suggestion
    def coords_to_point(self, *coords: ManimFloat):
```

This is what got me confused regarding my other comment

**Review by Viicos — APPROVED** [review]

**Pull request merged**